components/column-linking-table.vue

```diff
@@ -18,23 +19,138 @@
 
 <script>
 
+    // Allows for reference to store data by creating simple, implicit getters
+    // Fields listed in mapState below can be found in the store (index.js)
+    import { mapState } from "vuex";
+
+    // Allows for reference to store actions (index.js)
+    import { mapActions } from "vuex";
+
     export default {
 
         props: {
 
-            categoryClasses: { type: Object, required: true },
-            columnToCategoryMap: { type: Object, required: true },
-            fields: { type: Array, required: true },
-            selectedCategory: { type: String, required: true },
-            tableData: { type: Array, required: true }
+            selectedCategory: { type: String, required: true }
+        },
+
+        data() {
+
+            return {
+
+                columnToCategoryTable: [],
+
+                uiText: {
+
+                    tableFields: [
+
+                        { key: "column" },
+                        { key: "description" }
+                    ]
+                }
+            };
+        },
+
+        created() {
+
+            // Create the data structure for the category to column linking table
+            this.setupColumnToCategoryTable();
+        },
+
+        computed: {
+
+            ...mapState([
+
+                "categoryClasses",
+                "columnToCategoryMap",
+                "dataDictionary",
+                "dataTable"
+            ])
         },
 
         methods: {
 
+            ...mapActions([
```

Comment:
any reason these need to be actions instead of mutations? are you planning on using actions for any multi-mutation event? Or do you expect there to be some asynchronous logic to happen here?